src/solver_cvodes.cpp

```diff
@@ -1036,8 +1036,14 @@ static int fxdot(realtype t, N_Vector x, N_Vector xdot, void *user_data) {
     auto solver = dynamic_cast<CVodeSolver const *>(typed_udata->second);
     Expects(model);
 
-    if(solver->timeExceeded()) {
-        return AMICI_MAX_TIME_EXCEEDED;
+    // Avoid expensive syscalls by checking only every N rhs evaluations
+    static int eval_counter = 0;
+    int const interval = 1000;
+    if (++eval_counter / interval) {
```

Comment:
What about rather extending `solver->timeExceeded()` by adding the counter as private member of solver and checking against that in every call?

Reply:
Thought about it, but it wouldn't work with 

https://github.com/AMICI-dev/AMICI/blob/eba2cadcd4f9439c13cd19644351c6c1b7673c68/src/amici.cpp#L148

There we want to check on every call. Would have to add some force_check=True argument or a separate method.

Reply:
> Thought about it, but it wouldn't work with
> 
> https://github.com/AMICI-dev/AMICI/blob/eba2cadcd4f9439c13cd19644351c6c1b7673c68/src/amici.cpp#L148
> 
> There we want to check on every call. Would have to add some force_check=True argument or a separate method.

I think I would prefer that.